bin/setup

```diff
@@ -120,35 +120,28 @@ verify_bundled_gems (){
   } 2>&1
 }
 
-postgres_is_running (){
-  echo 'Checking that postgres is running'
+verify_yarn_packages (){
+  echo 'Updating Yarn packages...'
+
+  bundle exec yarn install 2>&1
 }
 
-verify_database_configuration (){
-  if [ -f config/database.yml ]
-  then
-    bundle exec rake db:create:all
+verify_database_migrations (){
+  echo 'Setting up database...'
+  migrate_existing_database || setup_database
```

Comment:
If a database already exists, the script will merely run any pending migrations. This should ensure we don't clobber our local data if we accidentally run `bin/setup` more than once.

If migration fails (most commonly in this scenario due to a missing database), then we assume this is a fresh installation and proceed with `db:setup`.